low-level/framework/utils/ipmi_client.py

```diff
@@ -161,39 +185,67 @@ def get_fru_list_by_type(self, fru_list, sensor_id_map):
                 for fru in fru_detail}
         return sensor_id_map
 
-    def _run_command(self, command, out_file=subprocess.PIPE):
-        """executes commands"""
-        process = subprocess.Popen(command, shell=True, stdout=out_file, stderr=subprocess.PIPE)
-        result = process.communicate()
-        return result, process.returncode
+    def _run_ipmitool_subcommand(self, subcommand, grep_args=None):
```

Comment:
`_run_ipmitool_subcommand()` is already in IpmiTool class implemented on top of IPMI class, I didn't get the statement `we need to avoid references using ipmitool directly here.`

Reply:
i assumed class name as IpmiClient only, taking from filename, my bad. Given this function with in ipmiTool class, we are good.